ray-operator/controllers/ray/common/pod.go

```diff
@@ -240,6 +244,20 @@ func getRayContainerIndex(pod v1.Pod) (index int) {
 	return 0
 }
 
+func getAutoscalerContainerIndex(pod v1.Pod) (autoscalerContainerIndex int) {
+	// we identify the autoscaler container based on its name
+	for i, container := range pod.Spec.Containers {
+		if container.Name == AutoscalerContainerName {
+			return i
+		}
+	}
+
+	// not found, use second container
+	// (This branch shouldn't be accessed -- the autoscaler container should be present.)
```

Comment:
Should we log an error or warning here, or possibly even exit the program if this happens?

Reply:
If this were a Python program, I'd add an assertion.

The most correct thing would probably be to bubble an error a couple layers up the call stack, but that's a bit too much work for something that's not logically possible.

I will add a log statement.

Reply:
Or panic, maybe.

Reply:
yeah, why don't we add a warning log statement both here and in the function above that we use the first or second container respectively :)

Reply:
Panic for autoscaler container, since it's supposed to be logically impossible.
Info statements for Ray container. IMO, using the first container should be the standard (also, that's the pattern suggested by the sample configs.)